core/templates/pages/exploration-player-page/services/question-player-engine.service.spec.ts

```diff
@@ -0,0 +1,1255 @@
+    let versionSpy =
+      spyOn(urlService, 'getExplorationVersionFromUrl').and.returnValue(1);
+
+    questionPlayerEngineService.init(
+      multipleQuestionBackendDict, successHandler, failHandler);
+
+    expect(questionPlayerSpy).toHaveBeenCalled();
+    expect(explorationIdSpy).toHaveBeenCalled();
+    expect(questionPlayerModeSpy).toHaveBeenCalled();
+    expect(versionSpy).toHaveBeenCalled();
+    expect(successHandler).toHaveBeenCalled();
+    expect(failHandler).not.toHaveBeenCalled();
+    expect(questionPlayerEngineService.getExplorationId())
+      .toBe('explorationId1');
+    expect(questionPlayerEngineService.questionPlayerMode).toBe(true);
+    expect(questionPlayerEngineService.getExplorationVersion()).toBe(1);
+    expect(questionPlayerEngineService.questions.length).toBe(3);
+  });
+
+  it('should update the exploration version if it' +
+    'is not in question player mode when initialized', fakeAsync(() => {
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+    let exploration = sampleExplorationDict;
+    exploration.version = 2;
+
+    spyOn(contextService, 'setQuestionPlayerIsOpen').and.returnValue(null);
+    spyOn(contextService, 'getExplorationId').and.returnValue('explorationId1');
+    spyOn(contextService, 'isInQuestionPlayerMode').and.returnValue(false);
+    spyOn(urlService, 'getExplorationVersionFromUrl').and.returnValue(1);
+
+    let readOnlySpy =
+      spyOn(readOnlyExplorationBackendApiService, 'loadExplorationAsync')
+        .and.callFake((expID, version) => {
+          expect(version).toBe(1);
+          return Promise.resolve(exploration);
+        });
+
+    questionPlayerEngineService.init(
+      multipleQuestionBackendDict, successHandler, failHandler);
+    tick();
+
+    expect(questionPlayerEngineService.questionPlayerMode).toBe(false);
+    expect(readOnlySpy).toHaveBeenCalled();
+    expect(questionPlayerEngineService.getExplorationVersion()).toBe(2);
+  }));
+
+  it('should return the current question', () => {
+    let successCallback = jasmine.createSpy('success');
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+    let exploration = sampleExplorationDict;
+    let answer = 'answer';
+    let interactionRulesService: InteractionRulesService;
+    let answerClassificationResult: AnswerClassificationResult = {
+      answerGroupIndex: 0,
+      classificationCategorization: 'default_outcome',
+      outcome: {
+        dest: null,
+        feedback: {
+          _contentId: 'feedback_id',
+          _html: '<p>Dummy Feedback</p>',
+          contentId: 'feedback_id',
+          html: '<p>Dummy Feedback</p>',
+        },
+        labelledAsCorrect: true,
+        missingPrerequisiteSkillId: null,
+        paramChanges: [],
+        refresherExplorationId: null,
+      },
+      ruleIndex: 0
+    } as unknown as AnswerClassificationResult;
```

Comment:
Why "as unknown" here and elsewhere?

Reply:
I was using workaround, but I found out a better way to that.
Updated.